src/main/java/org/mariadb/jdbc/internal/failover/impl/AuroraListener.java

```diff
@@ -154,47 +199,238 @@ public void reconnectFailedConnection(SearchFilter searchFilter) throws QueryExc
+        }
+
+    }
 
     /**
-     * Aurora replica doesn't have the master endpoint but the master instance name.
-     * since the end point normally use the instance name like "instancename.some_ugly_string.region.rds.amazonaws.com",
-     * if an endpoint start with this instance name, it will be checked first.
+     * Retrieves all endpoints of a cluster from the appropriate database table.
+     *
+     * @param protocol      current protocol connected to
+     * @return instance endpoints of the cluster
+     * @throws QueryException
+     */
+    private List<String> getCurrentEndpointIdentifiers(Protocol protocol) throws QueryException {
+        List<String> endpoints = new ArrayList<>();
+        try {
+            proxy.lock.lock();
+            try {
+                // Deleted instance may remain in db for 24 hours so ignoring instances that have had no change for IGNORE_TIME_IN_MINUTES
+                Date date = new Date();
+                int IGNORE_TIME_IN_MINUTES = 3;
+                Timestamp currentTime = new Timestamp(date.getTime() - IGNORE_TIME_IN_MINUTES*60*1000);
+                sqlDateFormat.setTimeZone(TimeZone.getTimeZone(protocol.getServerData("system_time_zone")));
+
+                SingleExecutionResult queryResult = new SingleExecutionResult(null, 0, true, false);
+                protocol.executeQuery(queryResult,
+                        "select server_id, session_id from " + dbName + ".replica_host_status " +
+                                "where last_update_timestamp > '" + sqlDateFormat.format(currentTime) + "'",
+                        ResultSet.TYPE_FORWARD_ONLY);
+                MariaSelectResultSet resultSet = queryResult.getResult();
+
+                while (resultSet.next()) {
+                    endpoints.add(resultSet.getString(1) + urlEndStr);
+                }
+
+            } finally {
+                proxy.lock.unlock();
+            }
+        } catch (SQLException se) {
+            log.log(Level.WARNING, "SQL exception occurred: " + se);
+        } catch (QueryException qe) {
+            if (protocol.getProxy().hasToHandleFailover(qe)) {
+                if (masterProtocol.equals(protocol)) {
+                    setMasterHostFail();
+                } else if (secondaryProtocol.equals(protocol)) {
+                    setSecondaryHostFail();
+                }
+                addToBlacklist(protocol.getHostAddress());
+                reconnectFailedConnection(new SearchFilter(isMasterHostFail(), isSecondaryHostFail()));
+            }
+        }
+
+        return endpoints;
+    }
+
+    /**
+     * Sets urlParser if there are any changes in the instance endpoints available.
+     *
+     * @param endpoints     instance identifiers
+     * @param clusterHost   host string of the cluster
+     * @param port          port that is common to all endpoints
+     */
+    private void setUrlParserFromEndpoints(List<String> endpoints, String clusterHost, int port) {
```

Comment:
Good point. There is no need since it is being stored in its own variable and used when necessary.